packages/peregrine/lib/talons/MagentoRoute/getRouteComponent.js

```diff
@@ -0,0 +1,56 @@
+import resolveUnknownRoute from '../../Router/resolveUnknownRoute';
+
+const getRouteComponent = async (apiBase, pathname) => {
+    // At build time, `fetchRootComponent` is injected as a global.
+    // Depending on the environment, this global will be either an
+    // ES module with a `default` property, or a plain CJS module.
+    const fetchRoot =
+        'default' in fetchRootComponent
+            ? fetchRootComponent.default
+            : fetchRootComponent;
+
+    try {
+        // try to resolve the route
+        // if this throws, we essentially have a 500 Internal Error
+        const resolvedRoute = await resolveUnknownRoute({
+            apiBase,
+            route: pathname
+        });
+
+        // urlResolver query returns null if a route can't be found
+        if (!resolvedRoute) {
+            throw new Error('404');
+        }
+
+        const { type, id } = resolvedRoute;
+        // if resolution and destructuring succeed but return no match
+        // then we have a straightforward 404 Not Found
+        if (!type || !id) {
+            throw new Error('404');
+        }
+
+        // at this point we should have a matching RootComponent
+        // if this throws, we essentially have a 500 Internal Error
+        const component = await fetchRoot(type);
+
+        // associate the matching RootComponent with this location
+        return {
+            component,
+            id,
+            pathname,
+            type
+        };
+    } catch (e) {
+        if (process.env.NODE_ENV === 'development') {
```

Comment:
I don't think we need this anymore. In prod mode all `console.*` statements are omitted.

Reply:
Fixed in [7a7ef14](https://github.com/magento/pwa-studio/pull/1966/commits/7a7ef14063db684a9103e63de7e826a78e08f1c3).